### PR TITLE
Blog: How to build registries

### PR DESCRIPTION
**Pull request by spara**

### CODE REVIEW

File: content/blog/how-to-registries/index.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    - Docker
+---
+
+Whether you are working with Kubernetes or serverless, your application uses containers. If you use the Docker desktop client, images are pushed to Docker Hub by default. Pulling images from Docker Hub is convenient, but there are many reasons to store images in your registry. For example, Docker Hub doesn’t guarantee to produce the same image on repeated pulls, i.e., your base image might have changed. It’s also possible to inadvertently expose secrets in an intermediate image used to build the image stored on Docker Hub. There is also the possibility of vulnerabilities in even official images. This article shows how to create a repository and how to build and push images to that repository
+
+<!--more-->
+
+## Container registry offerings
+
+The AWS provides the [Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/), Azure has  [Container Registry](https://azure.microsoft.com/en-us/services/container-registry/), and Google has it’s [Container Registry](https://cloud.google.com/container-registry). Each provider has associated services unique to their offering, but all support Docker or OCI compliant images.
```
Comment (stack72) [inline content/blog/how-to-registries/index.md:21]:
```suggestion
AWS provides the [Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/), Azure has  [Container Registry](https://azure.microsoft.com/en-us/services/container-registry/), and Google has it’s [Container Registry](https://cloud.google.com/container-registry). Each provider has associated services unique to their offering, but all support Docker or OCI compliant images.
```

File: content/blog/how-to-registries/index.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+Choose your cloud provider to learn how to build a registry.
+
+{{< chooser cloud "aws,azure,gcp" >}}
+{{< choosable cloud aws >}}
+
+In this example, we create an ECR repository configured to scan an image’s Operating System components. Scanning for vulnerabilities in an application is currently [out of scope](https://aws.amazon.com/blogs/containers/amazon-ecr-native-container-image-scanning/). We also set a policy for the repository that controls the actions allowed and a lifecycle policy that expires an image after a set time.
+
+With Pulumi, it’s possible to build an image locally using Docker and push it to your repository. To push the image, we obtain the credentials required to push from the registry. Finally, we export the credentials and the URL for the registry. Read more about ECR in the [API Reference]({{< relref "/docs/reference/pkg/aws/ecr" >}}).
+
+```typescript
```
Comment (stack72) [inline content/blog/how-to-registries/index.md:36]: Should we not have each language example?

**Review by stack72 — APPROVED** [review]
The post looks good - I left a small question about multi-language examples

File: content/blog/how-to-registries/index.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+meta_image: registry.png
+authors:
+    - sophia-parafina
+tags:
+    - containers
+    - registry
+    - Docker
+---
+
+Whether you are working with Kubernetes or serverless, your application uses containers. If you use the Docker desktop client, images are pushed to Docker Hub by default. Pulling images from Docker Hub is convenient, but there are many reasons to store images in your registry. For example, Docker Hub doesn’t guarantee to produce the same image on repeated pulls, i.e., your base image might have changed. It’s also possible to inadvertently expose secrets in an intermediate image used to build the image stored on Docker Hub. There is also the possibility of vulnerabilities in even official images. This article shows how to create a repository and how to build and push images to that repository
```
Comment (justinvp) [inline content/blog/how-to-registries/index.md:15]: "...there are many reasons to store images in your _**own**_ registry"

```suggestion
Whether you are working with Kubernetes or serverless, your application uses containers. If you use the Docker desktop client, images are pushed to Docker Hub by default. Pulling images from Docker Hub is convenient, but there are many reasons to store images in your own registry. For example, Docker Hub doesn’t guarantee to produce the same image on repeated pulls, i.e., your base image might have changed. It’s also possible to inadvertently expose secrets in an intermediate image used to build the image stored on Docker Hub. There is also the possibility of vulnerabilities in even official images. This article shows how to create a repository and how to build and push images to that repository
```

File: content/blog/how-to-registries/index.md (hunk 2 above)
Comment (spara) [reply content/blog/how-to-registries/index.md:36]: We should and I'll try to vary the language more in future blog posts.

**Pull request merged**